cpp/include/cudf/detail/stream_compaction.hpp

```diff
@@ -22,6 +22,7 @@
 #include <cudf/utilities/default_stream.hpp>
 
 #include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_uvector.hpp>
```

Comment:
This include doesn't look to be needed as no new types have been added with in the new function signature

Reply:
Oh yes, `device_uvector` is not needed for the new API but an old API uses it ([here](https://github.com/rapidsai/cudf/pull/11149/files/1c58baace3c8a062700061d10f0bff424ec9ff9e#diff-e0839e0c804747c1f5c3348574628ca8867766792cb9e6a9d0991e8872a7adb6R133)) so I add this header.